protocol/nats_jetstream/v3/protocol.go

```diff
@@ -0,0 +1,100 @@
+/*
+ Copyright 2021 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package nats_jetstream
+
+import (
+	"context"
+
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/cloudevents/sdk-go/v2/protocol"
+
+	"github.com/nats-io/nats.go"
+)
+
+// Protocol is a reference implementation for using the CloudEvents binding
+// integration. Protocol acts as both a NATS client and a NATS handler.
+type Protocol struct {
+	conn *nats.Conn
+
+	Consumer *consumer
```

Comment:
question: why do we need to keep those still public?

Reply:
in integration test:
```cleanup, s, r := testProtocol(ctx, t, conn, tt.args.consumerConfig, tt.args.opts...)```
the sender and receiver needs to be accesible from the protocol in order to call
```test.SendReceive(t, binding.WithPreferredEventEncoding(context.TODO(), tt.args.bindingEncoding), in, s, r,  ...```

I can add a GetSender() or GetConsumer() in order to get the values from the Protocol object.

Reply:
Hm, I wish there was a way we can keep this `internal` without external users of the SDK being able to access those fields/having to understand why they're public.

Reply:
I realized that Protocol IS a sender and IS a receiver.  So I was able to make them private